nodelet/include/nodelet/nodelet.h

```diff
@@ -127,6 +183,13 @@ class Nodelet
   // Protected data fields for use by the subclass.
 protected:
   inline const std::string& getName() const { return nodelet_name_; }
+  inline const std::string getSuffixedName(const std::string& suffix) const
+  {
+    std::string combined_name = nodelet_name_;
+    combined_name += std::string(".");
+    combined_name += suffix;
+    return combined_name;
```

Comment:
For code could be significantly simplified to:

    return nodelet_name_ + "." + suffix;